File: algorithms/easy/168_excel_sheet_column_title.py
# 168. Excel Sheet Column Title

def convert_to_title(column_number):
    """Convert a column number to the corresponding column title in Excel.
    
    In Excel, columns are labeled as A, B, C, ..., Z, AA, AB, ..., AZ, BA, ...
    which correspond to 1, 2, 3, ..., 26, 27, 28, ..., 52, 53, ...
    
    Args:
        column_number: Positive integer representing a column number
    
    Returns:
        String representing the corresponding column title
    """
    result = ""
    
    while column_number > 0:
        # Decrement by 1 to handle the 1-indexed nature of Excel columns
        column_number -= 1
        
        # Get the remainder when divided by 26 (number of letters in the alphabet)
        remainder = column_number % 26
        
        # Convert the remainder to the corresponding letter (A-Z) and add it to the result
        result = chr(65 + remainder) + result
        
        # Integer division by 26 to get the next digit
        column_number //= 26
    
    return result

# Test cases
print(convert_to_title(1))     # "A"
print(convert_to_title(28))    # "AB"
print(convert_to_title(701))   # "ZY"
print(convert_to_title(2147483647))  # "FXSHRXW"

File: algorithms/easy/168_excel_sheet_column_title.js
/**
 * 168. Excel Sheet Column Title
 * 
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function(columnNumber) {
    let result = "";
    
    while (columnNumber > 0) {
        // Decrement by 1 to handle the 1-indexed nature of Excel columns
        columnNumber--;
        
        // Get the remainder when divided by 26 (number of letters in the alphabet)
        const remainder = columnNumber % 26;
        
        // Convert the remainder to the corresponding letter (A-Z) and add it to the result
        result = String.fromCharCode(65 + remainder) + result;
        
        // Integer division by 26 to get the next digit
        columnNumber = Math.floor(columnNumber / 26);
    }
    
    return result;
};

// Test cases
console.log(convertToTitle(1));     // "A"
console.log(convertToTitle(28));    // "AB"
console.log(convertToTitle(701));   // "ZY"
console.log(convertToTitle(2147483647));  // "FXSHRXW"